host_core/lib/host_core/actors/actor_module.ex

```diff
@@ -202,6 +202,14 @@ defmodule HostCore.Actors.ActorModule do
     {:stop, :normal, :ok, agent}
   end
 
+  defp reconstitute_trace_context(headers) when is_list(headers) do
+    :otel_propagator_text_map.extract(headers)
+  end
+
+  defp reconstitute_trace_context(_) do
+    # NO OP
```

Comment:
Might be worth adding a warning or otherwise indicating if this unexpected case happens